examples/matching-engine/tests/transaction.rs

```diff
@@ -0,0 +1,263 @@
+
+    let fungible_bytecode_id_a = fungible_chain_a.publish_bytecodes_in("../fungible").await;
+    let fungible_bytecode_id_b = fungible_chain_b.publish_bytecodes_in("../fungible").await;
+
+    let initial_state_a =
+        InitialStateBuilder::default().with_account(owner_a, Amount::from_tokens(10));
+    let params_a = fungible::Parameters::new("A");
+    let token_id_a = fungible_chain_a
+        .create_application::<FungibleTokenAbi>(
+            fungible_bytecode_id_a,
+            params_a,
+            initial_state_a.build(),
+            vec![],
+        )
+        .await;
+    let initial_state_b =
+        InitialStateBuilder::default().with_account(owner_b, Amount::from_tokens(9));
+    let params_b = fungible::Parameters::new("B");
+    let token_id_b = fungible_chain_b
+        .create_application::<FungibleTokenAbi>(
+            fungible_bytecode_id_b,
+            params_b,
+            initial_state_b.build(),
+            vec![],
+        )
+        .await;
+
+    fungible_chain_a.register_application(token_id_b).await;
+    fungible_chain_b.register_application(token_id_a).await;
+
+    for (owner, amount) in [
+        (admin_account, None),
+        (owner_a, Some(Amount::from_tokens(10))),
+        (owner_b, None),
+    ] {
+        let value = FungibleTokenAbi::query_account(token_id_a, &fungible_chain_a, owner).await;
+        assert_eq!(value, amount);
+    }
+    for (owner, amount) in [
+        (admin_account, None),
+        (owner_a, None),
+        (owner_b, Some(Amount::from_tokens(9))),
+    ] {
+        let value = FungibleTokenAbi::query_account(token_id_b, &fungible_chain_b, owner).await;
+        assert_eq!(value, amount);
+    }
+
+    let tokens = [token_id_a, token_id_b];
+    let matching_parameter = Parameters { tokens };
+    let matching_id = matching_chain
+        .create_application::<MatchingEngineAbi>(
+            bytecode_id,
+            matching_parameter,
+            (),
+            vec![token_id_a.forget_abi(), token_id_b.forget_abi()],
+        )
+        .await;
+    // Doing the registrations
+    matching_chain.register_application(token_id_a).await;
+    matching_chain.register_application(token_id_b).await;
+    fungible_chain_a.register_application(matching_id).await;
+    fungible_chain_b.register_application(matching_id).await;
+
+    // Creating the bid orders
+    let mut orders_bids = Vec::new();
+    for price in [1, 2] {
+        let price = Price { price };
+        let order = Order::Insert {
+            owner: owner_a,
+            amount: Amount::from_tokens(3),
+            nature: OrderNature::Bid,
+            price,
+        };
+        let operation = Operation::ExecuteOrder { order };
+        let order_messages = fungible_chain_a
+            .add_block(|block| {
+                block.with_operation(matching_id, operation);
+            })
+            .await;
+
+        assert_eq!(order_messages.len(), 3);
```

Comment:
I don't remember why it's three: Sending the tokens, the order itself… what's the third?